pkg/util/validation/exporter.go

```diff
@@ -0,0 +1,85 @@
+package validation
+
+import (
+	"context"
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/cortexproject/cortex/pkg/util/services"
+)
+
+const typeTenant = "tenant"
+
+// OverridesExporter exposes per-tenant resource limit overrides as Prometheus metrics
+type OverridesExporter struct {
+	services.Service
+
+	overridesCh  <-chan map[string]*Limits
+	overridesMtx sync.RWMutex
+	overrides    map[string]*Limits
+
+	description *prometheus.Desc
+}
+
+// NewOverridesExporter creates an OverridesExporter that reads updates to per-tenant
+// limits from the provided channel. A single instance of the tenant to limit mapping
+// is kept in memory at a time by this exporter.
+func NewOverridesExporter(overridesCh <-chan map[string]*Limits) *OverridesExporter {
```

Comment:
Using function to access current limits map would simplify the design, and not require `OverridesExporter` to be a service. Have you considered that?

We could then reuse this function in overrides, and reduce amount of code in `runtime_config.go` -- single function will do just fine.

Reply:
:+1: I'll see what I can do